contracts/lib.rs

```diff
@@ -1672,1661 +1527,1788 @@ pub mod prosopo {
 
-        /// Test provider register and update
-        #[ink::test]
-        fn test_provider_register_and_update() {
-            let operator_accounts = get_operator_accounts();
-            let mut contract = Prosopo::default(
-                operator_accounts,
-                STAKE_DEFAULT,
-                STAKE_DEFAULT,
-                10,
-                1000000,
-                0,
-                1000,
-            );
-            let (provider_account, service_origin, fee) = generate_provider_data(0x2, "2424", 0);
-            ink::env::test::set_caller::<ink::env::DefaultEnvironment>(provider_account);
-            contract
-                .provider_register(service_origin, fee, Payee::Dapp)
-                .unwrap();
-            assert!(contract.providers.get(provider_account).is_some());
-            assert!(contract
-                .provider_accounts
-                .get(ProviderState {
-                    status: GovernanceStatus::Deactivated,
-                    payee: Payee::Dapp
-                })
-                .unwrap()
-                .contains(&provider_account));
-
-            let service_origin: Vec<u8> = vec![1, 2, 3];
-            let fee: u32 = 100;
-            ink::env::test::set_caller::<ink::env::DefaultEnvironment>(provider_account);
-            let balance = 20000000000000;
-            ink::env::test::set_value_transferred::<ink::env::DefaultEnvironment>(balance);
-            contract.provider_update(service_origin.clone(), fee, Payee::Dapp);
-            assert!(contract
-                .provider_accounts
-                .get(ProviderState {
-                    status: GovernanceStatus::Deactivated,
-                    payee: Payee::Dapp
-                })
-                .unwrap()
-                .contains(&provider_account));
-            let provider = contract.providers.get(provider_account).unwrap();
-            assert_eq!(provider.service_origin, service_origin);
-            assert_eq!(provider.fee, fee);
-            assert_eq!(provider.payee, Payee::Dapp);
-            assert_eq!(provider.balance, balance);
-            assert_eq!(provider.status, GovernanceStatus::Deactivated);
-        }
+                assert_eq!(
+                    contract.terminate(accounts.users[0]).unwrap_err(),
+                    Error::IsNotAdmin
+                );
+            }
 
-        /// Test provider register with service_origin error
-        #[ink::test]
-        fn test_provider_register_with_service_origin_error() {
-            let operator_accounts = get_operator_accounts();
-            let mut contract = Prosopo::default(
-                operator_accounts,
-                STAKE_DEFAULT,
-                STAKE_DEFAULT,
-                10,
-                1000000,
-                0,
-                1000,
-            );
+            // #[ink::test]
+            // fn test_withdraw() {
```

Comment:
Why is this commented?

Reply:
several of the unit tests are commented because I cannot find how to get the account of the contract reliably. Luckily, someone has answered my question on stack exchange about this and I think I know what's wrong now, so will be putting in a fix shortly

https://substrate.stackexchange.com/questions/8365/why-is-the-accountid-of-a-contract-always-1-when-unit-testing

Reply:
the withdraw test has now been fixed.

the set_code_hash test will not be fixed due to lack of own_code_hash() implementation